operators/src/source/ogr_source.rs

```diff
@@ -361,6 +364,50 @@ where
         }
     }
 
+    fn open_gdal_dataset(dataset_info: &OgrSourceDataset) -> Result<Dataset> {
+        // TODO: reliably detect CSV files or allow defining them as such in params
+        if let Some(ext) = dataset_info.file_name.extension() {
+            if ext == "csv" || ext == "tsv" {
+                let columns = dataset_info
+                    .columns
+                    .as_ref()
+                    .ok_or(error::Error::OgrSourceColumnsSpecMissing)?;
+
+                // TODO: make column x optional or allow other indication for data collection
+                if columns.x.is_empty() {
+                    return Ok(Dataset::open(&dataset_info.file_name)?);
+                }
+
+                if let Some(y) = &columns.y {
+                    return Ok(Dataset::open_ex(
+                        &dataset_info.file_name,
+                        None,
+                        None,
+                        Some(&[
+                            &("X_POSSIBLE_NAMES=".to_owned() + &columns.x),
+                            &("Y_POSSIBLE_NAMES=".to_owned() + y),
+                            "AUTODETECT_TYPE=YES",
+                        ]),
+                        None,
+                    )?);
+                }
+
+                return Ok(Dataset::open_ex(
+                    &dataset_info.file_name,
+                    None,
+                    None,
+                    Some(&[
+                        &("GEOM_POSSIBLE_NAMES=".to_owned() + &columns.x),
```

Comment:
I think you should use `format!` instead.

Reply:
992dbbe0c995dc1ba9aef4947614ba1dbd93eb96